src/timeline.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * Created by Martin on 04.01.2018.
+ */
+
+import {IMalevoDataset, IMalevoDatasetCollection, IMalevoEpochInfo} from './malevo_dataset';
+import * as d3 from 'd3';
+import * as events from 'phovea_core/src/event';
+import {AppConstants} from './app_constants';
+import * as ajax from 'phovea_core/src/ajax';
+import {INumericalMatrix} from 'phovea_core/src/matrix';
+import {IDragSelection} from './range_selector';
+import RangeSelector from './range_selector';
+
+export default class Timeline implements IDragSelection {
+  private readonly $node:d3.Selection<any>;
+  private $circles:d3.Selection<any>;
+  private $rubberband: d3.Selection<any>;
+  private isDragging = false;
+
+  constructor(parent: d3.Selection<any>) {
+    this.$node = parent.append('div').attr('id', 'timeline');
+    this.attachListener();
+  }
+
+  private attachListener() {
+    events.on(AppConstants.EVENT_DATA_COLLECTION_SELECTED, (evt, items:IMalevoDataset) => {
+     this.updateItems(items);
+    });
+  }
+
+  private getJSONEpochMetadata(data: INumericalMatrix) : Promise<any> {
+    return ajax.getAPIJSON(`/malevo_api/epoch/${data.desc.id}/ratio_bar`);
+  }
+
+  private loadConfusionData(malevoData: INumericalMatrix) : Promise<any> {
+    return malevoData.data()
+      .then((x) => {
+        console.log(x);
+      });
+  }
+
+  private createRubberband() {
+    this.$rubberband = this.$node.append('div').classed('selection', true);
+  }
+
+  updateItems(malevoData: IMalevoDataset) {
+    const that = this;
+    this.createRubberband();
+    const $circles = this.$node.selectAll('div.epochs')
+      .data(malevoData.epochInfos);
+
+    $circles.enter().append('div')
+      .classed('epochs', true)
+      .classed('loading', true)
+      .each(function(epochInfo: IMalevoEpochInfo, i: number) {
+        that.getJSONEpochMetadata(epochInfo.confusionInfo)
+          .then((json) => {
+            const $epochDiv = d3.select(this);
+            $epochDiv.classed('loading', false);
+            $epochDiv.html(`<div class="point"></div>
+                            <div class="label">${epochInfo.name}</div>`);
+            return epochInfo;
+          });
+      });
+    $circles.exit().remove();
+    this.$circles = $circles;
+    new RangeSelector(this.$node, this.$node.selectAll('div.epochs'), [this]);
```

Comment:
This will add an additional range selection with every call `updateItems()` --> I assume the initialization should happen in the `build()` function. 

Try to add a second dataset and see what happens.